### PR TITLE
k8s: Add KUBEVIRT_WITH_CNAO to provision cluster-network-addons-opera…

### PR DESCRIPTION
**Pull request by qinqon**
…tor at cluster-up

It also pre-pulls the CNAO images at provision and store the manifests at the VM so
everything is cached.

Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

**Comment by qinqon** [comment]
@mbarroso with this you will have k8s 1.17 with latest CNAO easily

File: cluster-up/cluster/k8s-provider-common.sh [hunk 1]
```diff
@@ -19,4 +19,17 @@ function up() {
...
     prepare_config
+
+    # Activate cluster-network-addons-operator if flag is passed
+    if [ "$KUBEVIRT_WITH_CNAO" == "true" ]; then
+        kubectl="${_cli} --prefix $provider_prefix ssh node01 -- sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf"
+
+        $kubectl create -f /tmp/cnao/namespace.yaml
+        $kubectl create -f /tmp/cnao/network-addons-config.crd.yaml
+        $kubectl create -f /tmp/cnao/operator.yaml
+        $kubectl wait deployment -n cluster-network-addons cluster-network-addons-operator --for condition=Available --timeout=200s
```
Comment (phoracek) [inline cluster-up/cluster/k8s-provider-common.sh]: Is Available condition available on a Deployment?

File: cluster-up/cluster/k8s-provider-common.sh [hunk 2]
```diff
@@ -19,4 +19,17 @@ function up() {
 
     # Make sure that local config is correct
     prepare_config
+
+    # Activate cluster-network-addons-operator if flag is passed
```
Comment (phoracek) [inline cluster-up/cluster/k8s-provider-common.sh]: Please add a documentation on how to use this

File: cluster-up/cluster/k8s-provider-common.sh (hunk 1 above)
Comment (qinqon) [reply cluster-up/cluster/k8s-provider-common.sh]: It has this conditions:
```yaml
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2020-03-03T16:03:12Z"
    lastUpdateTime: "2020-03-03T16:03:12Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    **type: Available**
  - lastTransitionTime: "2020-03-03T16:02:47Z"
    lastUpdateTime: "2020-03-03T16:03:12Z"
    message: ReplicaSet "cluster-network-addons-operator-69d7cd44f5" has successfully
      progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1

```

Comment (phoracek) [reply cluster-up/cluster/k8s-provider-common.sh]: :man_facepalming: I was blind.

**Pull request merged**